marshmallow_dataclass/__init__.py

```diff
@@ -268,6 +289,8 @@ def add_schema(_cls=None, base_schema=None, cls_frame=None, stacklevel=1):
     """
 
     def decorator(clazz: Type[_U], stacklevel: int = stacklevel) -> Type[_U]:
+        _check_decorated_type(clazz)
```

Comment:
I don't think so because we already have a big warning when a class is not a dataclass in `_internal_class_schema`
So non-dataclasses are allowed, but not supported

Reply:
I've just run a simple test.

You are correct in that decorating non-data class classes with `add_schema` works.  (That doesn't necessarily mean that we should allow it.)

It appears, however, that, as things stand in this PR, no big warning is emitted in that case.   Further investigation reveals that `get_resolved_dataclass_fields` "just works" (with no warnings emitted) even if its argument is not a dataclass.  (I have some suspicion that perhaps fields from the classes __mro__ are not properly handled in that case, but I haven't looked close enough to say for sure.)

In any case, I think that either:
- We should disallow using the `add_schema` decorator on non-dataclasses.  (Why allow it if it's unsupported/untested?)
- Or, we need a test to ensure the warning is, in fact, emitted when it is.

Reply:
I've had to do some digging.

1. In today's version, 8.7.1, I can call `class_schema(NonDataclass)` just fine. It only shows the warning if one of it's fields is not a dataclass. i.e.: Nested non dataclasses.
2. This goes back to when the warning was originally added: https://github.com/lovasoa/marshmallow_dataclass/commit/e31faa81a41691e8493ac18fb8f8aa44aeb52dbd
3. The behaviour still works the same with this PR.

I don't disagree with removing support for non-dataclasses, but don't see why that should be part of this PR.